src/routes/Router.tsx

```diff
@@ -26,10 +27,10 @@ type TRoutesWithSubRoutes = {
 const RouteWithSubRoutes = (route: TRoutesWithSubRoutes) => {
     return (
         <Switch>
-            {route.routes?.map(subRoute => (
-                <Route key={subRoute.path} path={subRoute.path} render={props => <subRoute.component {...props} />} />
-            ))}
-            <Route path={route.path} render={props => <route.component {...props} />} />
+            {/* Render subroutes recursively with any number of nested routes */}
+            {route.routes?.map(subRoute => <RouteWithSubRoutes key={subRoute.path} {...subRoute} />)}
+            <Route component={route.component} exact path={route.path} />
+            <Route component={Page404} />
```

Comment:
do we need change this? wont it by default if theres no match for the routes, fall to the last Route below where the Page404 is?

Reply:
no.. it doesn't.. for the other change it goes to 404 .. but that doesn't handle nested routes.. so we need this for the nested ones..

Reply:
weird, it should fallback by default, but alright 👍 so all the nested routes are ok? did you check if it affects those routes like completing orders, share my ads

Reply:
yes. those are working fine.